CeresDB/ceresmeta
# refactor: optimize hash node select algo

# Which issue does this PR close?

Closes #200  #208 

# Rationale for this change
As mentioned in the issue, we need to optimize the current algorithm for assigning shards to nodes so that the distribution of shards on nodes becomes more even.

# What changes are included in this PR?
* Replace `ConsistentHashRing` with `UniformityHash`.
* Refactor `NodePicker` based on `uniformityHash`, which make the distribution of shards on nodes becomes more even.

# Are there any user-facing changes?
* The distribution of shards on nodes becomes more even.

# How does this change test
Pass all unit tests and integration test, and observe the distribution of Shard in the unit test to ensure that its distribution is reasonable.
